tests/unit/http/test_http_client.py

```diff
@@ -42,6 +42,61 @@ def test_request_sets_host_header_if_missing(self):
         self.assertIsNotNone(self.client.last_request)
         self.assertIsNotNone(self.client.last_response)
 
+    def test_request_with_timeout(self):
+        self.request_mock.url = 'https://api.twilio.com/'
+        self.request_mock.headers = {'Host': 'other.twilio.com'}
+
+        response = self.client.request(
+            'doesnt matter', 'doesnt matter', None, None, None, None, 30, None)
+
+        self.assertEqual('other.twilio.com', self.request_mock.headers['Host'])
+        self.assertEqual(200, response.status_code)
+        self.assertEqual('testing-unicode: Ω≈ç√, 💩', response.content)
+
+    def test_request_where_method_timeout_equals_zero(self):
+        self.request_mock.url = 'https://api.twilio.com/'
+        self.request_mock.headers = {'Host': 'other.twilio.com'}
+
+        try:
+            self.client.request(
+                'doesnt matter', 'doesnt matter', None, None, None, None, 0, None)
+        except ValueError as e:
+            self.assertEqual(ValueError, type(e))
```

Comment:
This may be redundant since `e` could only ever be a `ValueError`. I think it'd be more beneficial to test that when able to catch any exceptions, then we end up getting a ValueError type in this case.
```python
except Exception as e:
    self.assertEqual(ValueError, type(e))
```